config/zephyr/Kconfig

```diff
@@ -95,6 +95,7 @@ config CHIP_DEVICE_HARDWARE_VERSION_STRING
 	  hardware version string is exposed as an attribute of the Basic
 	  Information cluster.
 
+if "$(APPVERSION)"=""
```

Comment:
Just a thought.. what about to use `depends` attribute to below KConfig instead of if-statement? In this case user can get warning during building when using both versioning methods. Or its worlaround to fix other issue? ;)

Reply:
I think that "if" block is change into `depends` statement for every config inside, during the build process.